ci/gh-build.sh

```diff
@@ -27,7 +27,7 @@ set -euo pipefail
 set -x
 
 # First, basic static analysis
-./ci/codestyle.sh
+./ci/codestyle.sh clang-format
```

Comment:
What does this argument do?

Reply:
I can remove the "clang-format" arg if needs be. I added it because I felt the ci/codestyle.sh shouldn't do git operations like fetch, rebase, etc. and apply reformats by default. But it's the only way I could thing of, of getting git clang-format to just look at the changes in a PR. If we reformat the whole codebase though, you can simply call clang-format to check the whole codebase.

Another thing I've seen happen, just to be mindful of if we start seeing it, is sometimes you have to do version checks eventually like only check if clang-format version is greater than 12. Because sometimes how it reformats between versions can change slightly, and you get into the situation where so some builds thing the style is ok, and some don't depending on clang-format version.

This is the way I did it in inotify-tools CI, is we only support clang-format greater than 11 because earlier versions reformat a little different:

https://github.com/inotify-tools/inotify-tools/blob/8367bdd1eda71c10ab0a49c694846a622e543807/build_and_test.sh#L80

I didn't reformat the whole codebase in inotify-tools. Like I am proposing here, all new changes to inotify-tools are mandated to correspond with the .clang-format style. It avoids the "git blame" being confusing problem.